x/mongo/driver/topology/server.go

```diff
@@ -649,7 +651,9 @@ func (s *Server) updateDescription(desc description.Server) {
 // createConnection creates a new connection instance but does not call connect on it. The caller must call connect
 // before the connection can be used for network operations.
 func (s *Server) createConnection() (*connection, error) {
-	opts := []ConnectionOption{
+	opts := make([]ConnectionOption, len(s.cfg.connectionOpts))
+	copy(opts, s.cfg.connectionOpts)
+	opts = append(opts,
 		WithConnectTimeout(func(time.Duration) time.Duration { return s.cfg.heartbeatTimeout }),
```

Comment:
Should these options refer to `opts` instead of `s.cfg`?

Now that `s.cfg` is no longer overwritten, I do not think it is necessary. But since `s.cfg` is being copied, it seems natural to use the copy.

Reply:
Maybe I'm misunderstanding `copy`, but why do we need this `append` statement anymore? Shouldn't the `copy` on L655 already have copied these sub-options?

Reply:
My understanding is that the appended `ConnectionOption`s aren't already in the `s.cfg.connectionOpts` slice and the intent is to convert options on the `s.cfg` struct to options in the `[]ConnectionOption` so they can be used to create a new connection.

Reply:
Gotcha, that makes sense.

Reply:
> Are you referring to the other configuration values on `s.cfg`, like `s.cfg.heartbeatTimeout`? 

Yes, I misunderstood this. Thank you for clarifying.